libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/factory/RouteLegProgressFactory.kt

```diff
@@ -0,0 +1,51 @@
+package com.mapbox.navigation.base.internal.factory
+
+import com.mapbox.api.directions.v5.models.LegStep
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.navigation.base.trip.model.RouteLegProgress
+import com.mapbox.navigation.base.trip.model.RouteStepProgress
+
+/**
+ * Internal factory to build [RouteLegProgress] objects
+ */
+object RouteLegProgressFactory {
```

Comment:
Could you add the `ExperimentalMapboxNavigationAPI` to this class?

Reply:
Adding this annotation has a ripple effect to `MapboxTripSession` and other classes.  `OptIn` requires a compiler argument which I assume we want to avoid.

Reply:
That shouldn't be a problem to have them in our code and it's important that _someone else_ using this factory is notified by the compiler that they probably shouldn't.

Reply:
I started adding the annotation and it goes out to `MapboxNavigation` via `NavigationComponentProvider` since it creates a `MapboxTripSession`. It also touches `MapboxBuildingArrivalApi` and others.  Every navigation app. is going to instantiate `MapboxNavivation` and thus will have to deal with this annotation. Is this really what you intended?

Reply:
ok, done